eppo_core/src/eval/eval_precomputed_assignments.rs

```diff
@@ -2,29 +2,58 @@ use std::collections::HashMap;
 use std::sync::Arc;
 
 use chrono::{DateTime, Utc};
+use serde::{Deserialize, Serialize};
 
 use crate::eval::get_assignment;
-use crate::ufc::Assignment;
-use crate::ufc::{AssignmentFormat, Environment};
-use crate::{Attributes, Configuration, EvaluationError, Str};
+use crate::ufc::{Assignment, AssignmentFormat, Environment, ValueWire, VariationType};
+use crate::{Attributes, Configuration, Str};
 
-#[derive(Debug)]
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
 pub struct PrecomputedConfiguration {
-    pub created_at: DateTime<Utc>,
-    pub format: AssignmentFormat,
-    pub environment: Environment,
-    pub flags: HashMap<String, Result<Assignment, EvaluationError>>,
+    created_at: DateTime<Utc>,
+    /// `format` is always `AssignmentFormat::Precomputed`.
+    format: AssignmentFormat,
+    // Environment might be missing if configuration was absent during evaluation.
+    environment: Option<Environment>,
+    flags: HashMap<String, PrecomputedAssignment>,
 }
 
-impl PrecomputedConfiguration {
-    pub fn empty(environment_name: &str) -> Self {
-        Self {
-            created_at: Utc::now(),
-            format: AssignmentFormat::Precomputed,
-            environment: Environment {
-                name: environment_name.into(),
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PrecomputedAssignment {
+    variation_type: VariationType,
+    variation_value: ValueWire,
```

Comment:
One diff here is that JSON assignment are now encoded as strings — following UFC flags config approach